### ISSUE TITLE
Ansible breaks backwards-compatibility when constructing paths for source files

### ISSUE THREAD

**Issue by cherti** [ansible/ansible#17877]
##### ISSUE TYPE
- Bug Report, Regression
##### COMPONENT NAME

tasks-execution, path-creation, to be observed at least in the `file`\- and `template`-module
##### ANSIBLE VERSION

```
ansible 2.2.0
```
##### CONFIGURATION

Central ansible-folder changed to ~/.ansible
roles-folder configured to ~/.ansible/roles
##### OS / ENVIRONMENT

Linux, but most likely unrelated to OS
##### SUMMARY

`template: src=template/{{ item }} dest=/etc/{{ item }}` used to work in roles
`file: src=template/{{ item }} dest=/etc/{{ item }}` used to work as well in roles.

(with_items-section deliberately left out).

With the above mentioned version, this must be changed to 
`template: src={{ item }} dest=/etc/{{ item }}` to work.
##### STEPS TO REPRODUCE

create a role with files

```
templates/foobar.txt
tasks/main.yml
```

in the main.yml, put:

```
- name: demonstration
  template: src=templates/foobar.txt dest=/etc/foobar.txt
```

and execute that. With ansible 2.1 this works, with the above noted version, it doesn't.

Error: `"Unable to find 'templates/foobar.txt' in expected paths."`

Changing to 

```
- name: demonstration
  template: src=foobar.txt dest=/etc/foobar.txt
```

fixes the issue for the templating module.

**Comment by agaffney** [comment]
I ran into this in 2.2.0 RC1 using the 'copy' module. The following task:

```
- name: .vimrc
  copy:
    dest: "{{ homedir }}/.vimrc"
    src: files/vimrc
    owner: "{{ owner }}"
    mode: "{{ default_file_mode }}"
```

with the source file at `roles/<role>/files/vimrc` results in:

```
fatal: [nuc1]: FAILED! => {"changed": false, "failed": true, "msg": "Unable to find 'files/vimrc' in expected paths."}
```

**Comment by agaffney** [comment]
Debug output:

```
  5126 1476327940.53620: search_path:
        /home/agaffney/dev/ansible_stuff/roles/homedir/files/files/vimrc
        /home/agaffney/dev/ansible_stuff/roles/homedir/tasks/files/vimrc
        /home/agaffney/dev/ansible_stuff/roles/homedir/tasks/files/files/vimrc
        /home/agaffney/dev/ansible_stuff/roles/homedir/tasks/tasks/files/vimrc
        /home/agaffney/dev/ansible_stuff/files/files/vimrc
        /home/agaffney/dev/ansible_stuff/files/vimrc
```

**Comment by agaffney** [comment]
I created a PR to fix this, but it may not be "correct".

https://github.com/ansible/ansible/pull/17995

**Comment by cherti** [comment]
is this fix included 2.2? Because this just hit me with the new ansible 2.2 with the copy-module. As I'm not sure if I should open another issue for this or if it is this code not having landed in 2.2, I thought I might ask here. :)

**Comment by agaffney** [comment]
Yes, this is fixed as of 2.2.0 RC2. It's likely not the same issue, as I can verify it's still fixed in 2.2.0

**Comment by cherti** [comment]
then I'll open another issue for this.
